fmriprep/workflows/base.py

```diff
@@ -374,77 +382,113 @@ def init_single_subject_wf(subject_id, task_id, name,
-                                               use_syn=use_syn,
-                                               force_syn=force_syn,
                                                debug=debug,
-                                               output_grid_ref=output_grid_ref,
-                                               use_aroma=use_aroma,
-                                               ignore_aroma_err=ignore_aroma_err)
+                                               longitudinal=longitudinal,
+                                               omp_nthreads=omp_nthreads,
+                                               freesurfer=freesurfer,
+                                               hires=hires,
+                                               reportlets_dir=reportlets_dir,
+                                               output_dir=output_dir)
 
         workflow.connect([
-            (anat_preproc_wf, func_preproc_wf,
-             [('outputnode.t1_preproc', 'inputnode.t1_preproc'),
-              ('outputnode.t1_brain', 'inputnode.t1_brain'),
-              ('outputnode.t1_mask', 'inputnode.t1_mask'),
-              ('outputnode.t1_seg', 'inputnode.t1_seg'),
-              ('outputnode.t1_tpms', 'inputnode.t1_tpms'),
-              ('outputnode.t1_2_mni_forward_transform', 'inputnode.t1_2_mni_forward_transform'),
-              ('outputnode.t1_2_mni_reverse_transform', 'inputnode.t1_2_mni_reverse_transform'),
-              # Undefined if --no-freesurfer, but this is safe
-              ('outputnode.subjects_dir', 'inputnode.subjects_dir'),
-              ('outputnode.subject_id', 'inputnode.subject_id'),
-              ('outputnode.t1_2_fsnative_forward_transform',
-               'inputnode.t1_2_fsnative_forward_transform'),
-              ('outputnode.t1_2_fsnative_reverse_transform',
-               'inputnode.t1_2_fsnative_reverse_transform')]),
+            (inputnode, anat_preproc_wf, [('subjects_dir', 'inputnode.subjects_dir')]),
+            (bidssrc, bids_info, [(('t1w', fix_multi_T1w_source_name), 'in_file')]),
+            (inputnode, summary, [('subjects_dir', 'subjects_dir')]),
+            (bidssrc, summary, [('t1w', 't1w'),
+                                ('t2w', 't2w'),
+                                ('bold', 'bold')]),
+            (bids_info, summary, [('subject_id', 'subject_id')]),
+            (bidssrc, anat_preproc_wf, [('t1w', 'inputnode.t1w'),
+                                        ('t2w', 'inputnode.t2w')]),
+            (summary, anat_preproc_wf, [('subject_id', 'inputnode.subject_id')]),
+            (bidssrc, ds_summary_report, [(('t1w', fix_multi_T1w_source_name), 'source_file')]),
+            (summary, ds_summary_report, [('out_report', 'in_file')]),
+            (bidssrc, ds_about_report, [(('t1w', fix_multi_T1w_source_name), 'source_file')]),
+            (about, ds_about_report, [('out_report', 'in_file')]),
+        ])
+
+        if anat_only:
+            return workflow
```

Comment:
If I specify `--anat-only` and `--func-only`, this won't be hit, so the functional workflows will still be generated. There should probably at some point (possibly in `run.py`) be a check to raise an exception if both flags are passed to fmriprep.

Reply:
Ok, I would just put in a simple if anat_only and func_only raise exception before any of the workflows are put together. Any reasons to put it elsewhere?

Reply:
The "proper" way to do this is using [mutually exclusive options](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group)

```Python
    # Create a mutually exclusive group for --anat-only and --func-only
    g_blocks = parser.add_mutually_exclusive_group()
    g_blocks.add_argument('--anat-only', action='store_true',
                          help='run anatomical workflows only')
    g_blocks.add_argument('--func-only', action='store_true',
                          help='run functional workflows only '
                               '(dismiss the anatomical images if present).')
```